contracts/upgradeable_contracts/RTokenConnector.sol

```diff
@@ -0,0 +1,51 @@
+pragma solidity 0.4.24;
+
+import "openzeppelin-solidity/contracts/AddressUtils.sol";
+import "../interfaces/IRToken.sol";
+import "./Ownable.sol";
+import "./ERC20Bridge.sol";
+
+contract RTokenConnector is Ownable, ERC20Bridge {
+    bytes32 internal constant RTOKEN = 0xdce7cb71f7466970418465f97f1fb7d721c1661bc59b1f0caa7d1f8ce374a47e; // keccak256(abi.encodePacked("rToken"))
+
+    function initializeRToken(
+        address _rToken,
+        address[] _recipients,
+        uint32[] _proportions
+    ) external onlyOwner returns (uint256) {
+        require(AddressUtils.isContract(_rToken));
+        addressStorage[RTOKEN] = _rToken;
+        erc20token().approve(_rToken, uint256(-1));
```

Comment:
Is it safe to approve operations for maximum amount of tokens? Does it make sense to approve the corresponding amount of tokens as part of the `mintRToken` operation?

Reply:
Agree with you. I thought about lower costs for transactions but this change won't make much difference